### PR TITLE
Review Dockerfile for building odh-mlmd-grpc-server in konflux

### PR DESCRIPTION
**Pull request by MohammadiIram**
Resolves #https://issues.redhat.com/browse/RHOAIENG-14224

### CODE REVIEW

File: Dockerfile.konflux [hunk 1]
```diff
@@ -0,0 +1,72 @@
+FROM quay.io/bcook/bazel:ubi8-bazel-5 as builder
+
+USER root
+
+RUN dnf update -y -q && \
+  dnf install -y -q \
+  which \
```
Comment (zdtsw) [inline Dockerfile.konflux:7]: why need install `which` ?

File: Dockerfile.konflux [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  openssl \
+  ca-certificates \
+  unzip \
+  git \
+  findutils \
+  python3
+
+COPY . /mlmd-src
+WORKDIR /mlmd-src
+
```
Comment (zdtsw) [inline Dockerfile.konflux:22]: good to have 
`RUN bazel version `
for debug purpose

File: Dockerfile.konflux [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+CMD \
+  "/bin/metadata_store_server" \
+  "--grpc_port=${GRPC_PORT}" \
+  "--metadata_store_server_config_file=${METADATA_STORE_SERVER_CONFIG_FILE}"
+
+
+ENTRYPOINT \
+  "/bin/metadata_store_server" \
+  "--grpc_port=${GRPC_PORT}" \
+  "--metadata_store_server_config_file=${METADATA_STORE_SERVER_CONFIG_FILE}"
```
Comment (zdtsw) [inline Dockerfile.konflux:59]: why add ENTRYPOINT if already CMD set ?

File: Dockerfile.konflux [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+COPY . /mlmd-src
+WORKDIR /mlmd-src
+
+# Running in offline mode with --nofetch arg, cache and deps must be cloned 
+# into the local root bazel cache
+# "-std=c++17" is needed in order to build with ZetaSQL.
+RUN bazel build -c opt --action_env=PATH \
+  --define=grpc_no_ares=true \
+  //ml_metadata/metadata_store:metadata_store_server \
+  --cxxopt="-std=c++17" --host_cxxopt="-std=c++17"
```
Comment (zdtsw) [inline Dockerfile.konflux:29]: guess no harm to keep -`-verbose_failures`

**Review by Al-Pragliola — COMMENTED** [review]
Couldn't we just use the content of https://github.com/red-hat-data-services/ml-metadata/blob/konflux-2.16/ml_metadata/tools/docker_server/Dockerfile.redhat , and add the labels?

File: Dockerfile.konflux [hunk 5]
```diff
@@ -0,0 +1,72 @@
+FROM quay.io/bcook/bazel:ubi8-bazel-5 as builder
```
[the remark this answers is not in the document]
Comment (MohammadiIram) [reply Dockerfile.konflux:1]: Currently, we are using this image, but we will switch to the official version of this Bazel image once it's available by Konflux team.

File: Dockerfile.konflux (hunk 4 above)
Comment (MohammadiIram) [reply Dockerfile.konflux:29]: Done

**Comment by MohammadiIram** [comment]
> Couldn't we just use the content of https://github.com/red-hat-data-services/ml-metadata/blob/konflux-2.16/ml_metadata/tools/docker_server/Dockerfile.redhat , and add the labels?

Yes, the contents are same with added LABELs

**Review by Al-Pragliola — DISMISSED** [review]
LGTM

My only concern is if konflux has the same requirements that CPaaS had, like having all dependencies in a tar.gz and so on

update: Removing the LGTM as I'll test with bazel build --nofetch for network isolated builds

**Comment by zdtsw** [comment]
> ~LGTM~
> 
> My only concern is if konflux has the same requirements that CPaaS had, like having all dependencies in a tar.gz and so on
> 
> update: Removing the LGTM as I'll test with bazel build --nofetch for network isolated builds, @zdtsw wdyt?

i heard, i am not 100%,  there will be some cached stuff done by konflux but i dont know the detail.
on the safe said, you can do air-gapped build with this dockerfile,  just to see how it goes.

as for the first base image, i am not sure if that will work with konflux in reality. i leave it to  @MohammadiIram or @dchourasia to figure it out

**Pull request merged**